### DISCUSSION TITLE
I need help for the Modchart

### DISCUSSION THREAD

**Discussion by LukasRainbow** [KadeArchive/Kade-Engine#2321] [Q&A]
I have a few questions about coding the Modchart...

> **1. How do i make the Arrows turn faster/further?**
>
>The Arrows so far only turn very slightly left and right. Is there a way to turn them further?

>**2. How do i move Arrows only in one Direction?**
>
>The Arrows either Move in Circles, or don´t move at all, but i want them to only go along one axis (in this case only the x axis, aka horizontally). How do i do that?

>**3. How do i Activate Downscroll, Show only Charting and HP Bar, and change the Window´s Size?**
>
>For some i know the Code but don´t know where to put it. For some i just have no idea where to start. Someone help plz?

Here is the Code i have so far:

```
function start(song)
    spinLength = 10
end


function update(elapsed)

    if difficulty == 2 and curStep > 1 then
        if spinLength < 20 then
            spinLength = spinLength + 25
        end


        local currentBeat = (songPos / 1000)*(bpm/60)
	for i=0,7,1 do
            local receptor = _G['receptor_'..i]
            receptor.angle = (spinLength / 7) * -math.sin((currentBeat + i*10) * math.pi)
	    receptor.x = receptor.defaultX + spinLength * math.sin((currentBeat + i*50) * math.pi)
	    receptor.y = receptor.defaultY + spinLength * math.cos((currentBeat + i*0.3) * math.pi)
        end
    end
end

function playerTwoTurn()
    camGame:tweenZoom(1,(crochet * 4) / 1000)
end

function playerOneTurn()
    camGame:tweenZoom(1,(crochet * 4) / 1000)
end
```

If any of you know how to answer these Two questions, let me know!

**Comment by glowsoony** [comment]
faster? well you need to change the code dude my code is this you can copy it or use it! they 

function start (song)
	print("Song: " .. song .. " @ " .. bpm .. " downscroll: " .. downscroll)

end


function update (elapsed)
    local currentBeat = (songPos / 1000)*(bpm/180)
    if curStep >= 259 and curStep < 494 then
        for i=0,8 do
            setActorY(_G['defaultStrum'..i..'Y'] + 5 * math.sin((currentBeat + i*0.90) * math.pi), i)
        end
    end

    if curStep >= 494 and curStep < 721 then
        for i=0,8 do
            setActorY(_G['defaultStrum'..i..'Y'] - 9 * math.sin((currentBeat + i*0.25) * math.pi), i)
        end
    end

    if curStep >= 721 and curStep < 1066 then
        for i=0,8 do
            setActorX(_G['defaultStrum'..i..'X'] + 20 * math.sin((currentBeat + i*0.90) * math.pi), i)
        end
    end

    if curStep >= 1066 and curStep < 1181 then
        for i=0,8 do
            setActorX(_G['defaultStrum'..i..'X'] - 32 * math.sin((currentBeat + i*0.25) * math.pi), i)
        end
    end

    if curStep >= 1181 and curStep < 1405 then
        for i=0,8 do
            setActorX(_G['defaultStrum'..i..'X'] + 38 * math.sin((currentBeat + i*0.25) * math.pi), i)
        end
    end

    if curStep >= 1405 and curStep < 1641 then
        for i=0,8 do
            setActorY(_G['defaultStrum'..i..'Y'] - 34 * math.sin((currentBeat + i*0.16) * math.pi), i)
        end
    end

    if curStep >= 1641 and curStep < 1681 then
        for i=0,8 do
            setActorX(_G['defaultStrum'..i..'X'] + 40 * math.sin((currentBeat + i*0.35) * math.pi), i)
        end
    end

    if curStep >= 1681 and curStep < 1689 then
        for i=0,8 do
            setActorY(_G['defaultStrum'..i..'Y'] - 5 * math.sin((currentBeat + i*0.9) * math.pi), i)
        end
    end

    if curStep >= 1689 and curStep < 1762 then
        for i=0,8 do
            setActorX(_G['defaultStrum'..i..'X'] + 50 * math.sin((currentBeat + i*0.55) * math.pi), i)
        end
    end

    if curStep >= 1762 and curStep < 1778 then
        for i=0,8 do
            setActorY(_G['defaultStrum'..i..'Y'] - 8 * math.sin((currentBeat + i*0.05) * math.pi), i)
        end
    end

    if curStep >= 1778 and curStep < 1875 then
        for i=0,8 do
            setActorX(_G['defaultStrum'..i..'X'] + 10 * math.sin((currentBeat + i*0.15) * math.pi), i)
        end
    end

    if curStep >= 1875 and curStep < 1900 then
        for i=0,8 do
            setActorY(_G['defaultStrum'..i..'Y'] - 70 * math.sin((currentBeat + i*0.95) * math.pi), i)
        end
    end

    if curStep >= 1066 and curStep < 1778 then
        tweenCameraZoom(1.2, 0.5)
    end

    if curStep >= 1778 and curStep < 1181 then
        tweenCameraZoom(1.2, 0.5)
    end


end



function beatHit (beat)
   -- do nothing

end 

function stepHit (step)
 
if curStep >= 1181 then
        dad:tweenAlpha(0.8,0.4)
  end

if curStep >= 1181 then
        dad:tweenAlpha(0,0.4)
   end
end

function keyPressed (key)
   -- do nothing
end

print("Mod Chart script loaded :)")

**Comment by glowsoony** [comment]
for downscroll i dont really know what code to use.
